### ISSUE TITLE
Dumps core all the time

### ISSUE THREAD

**Issue by andruska** [J-Lentz/iwgtk#38]
Only starts in root privileges. When running as regular non-root user:

```
$ iwgtk

(iwgtk:26654): GLib-WARNING **: 01:19:34.761: GError set over the top of a previous GError or uninitialized memory.
This indicates a bug in someone's code. You must ensure an error is NULL before it's set.
The overwriting error message was: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: Rejected send message, 2 matched rules; type="method_call", sender=":1.174" (uid=60139 pid=26654 comm="iwgtk") interface="org.freedesktop.DBus.ObjectManager" member="GetManagedObjects" error name="(unset)" requested_reply="0" destination=":1.10" (uid=0 pid=328 comm="/usr/lib/iwd/iwd")
Error creating GDBusObjectManager: 
Segmentation fault (core dumped)
```

When running as root:
```
# iwgtk

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.745: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.745: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.745: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.745: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.745: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.746: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.746: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.755: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.755: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.789: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.791: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.792: g_dbus_connection_emit_signal: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed

(iwgtk:27036): GLib-GIO-CRITICAL **: 01:31:25.847: g_dbus_connection_register_object: assertion 'object_path != NULL && g_variant_is_object_path (object_path)' failed
```

And evetually:

```
nov   20 01:19:34 ebook dbus-daemon[327]: [system] Rejected send message, 2 matched rules; type="method_call", sender=":1.174" (uid=60139 pid=26654 comm="iwgtk") interface="org.>
nov   20 01:19:34 ebook audit[26654]: ANOM_ABEND auid=60139 uid=60139 gid=60139 ses=2 pid=26654 comm="iwgtk" exe="/usr/bin/iwgtk" sig=11 res=1
nov   20 01:19:34 ebook kernel: traps: iwgtk[26654] general protection fault ip:7f406b4e482b sp:7ffe5d968c40 error:0 in libglib-2.0.so.0.7400.1[7f406b4af000+93000]
nov   20 01:19:34 ebook kernel: audit: type=1701 audit(1668899974.756:212): auid=60139 uid=60139 gid=60139 ses=2 pid=26654 comm="iwgtk" exe="/usr/bin/iwgtk" sig=11 res=1
nov   20 01:19:35 ebook systemd-coredump[26669]: [🡕] Process 26654 (iwgtk) of user 60139 dumped core.
                                                     
                                                     Stack trace of thread 26654:
                                                     #0  0x00007f406b4e482b n/a (libglib-2.0.so.0 + 0x5282b)
                                                     #1  0x00007f406b4e79b0 g_main_context_dispatch (libglib-2.0.so.0 + 0x559b0)
                                                     #2  0x00007f406b53e279 n/a (libglib-2.0.so.0 + 0xac279)
                                                     #3  0x00007f406b4e6132 g_main_context_iteration (libglib-2.0.so.0 + 0x54132)
                                                     #4  0x00007f406b7100ce g_application_run (libgio-2.0.so.0 + 0xde0ce)
                                                     #5  0x00005612754a6171 n/a (iwgtk + 0x7171)
                                                     #6  0x00007f406b2ce290 n/a (libc.so.6 + 0x23290)
                                                     #7  0x00007f406b2ce34a __libc_start_main (libc.so.6 + 0x2334a)
                                                     #8  0x00005612754a61b5 n/a (iwgtk + 0x71b5)
                                                     
                                                     Stack trace of thread 26656:
                                                     #0  0x00007f406b4e2188 n/a (libglib-2.0.so.0 + 0x50188)
                                                     #1  0x00007f406b4e4830 n/a (libglib-2.0.so.0 + 0x52830)
                                                     #2  0x00007f406b4e79b0 g_main_context_dispatch (libglib-2.0.so.0 + 0x559b0)
                                                     #3  0x00007f406b53e279 n/a (libglib-2.0.so.0 + 0xac279)
                                                     #4  0x00007f406b4e6ddf g_main_loop_run (libglib-2.0.so.0 + 0x54ddf)
                                                     #5  0x00007f406b73c33c n/a (libgio-2.0.so.0 + 0x10a33c)
                                                     #6  0x00007f406b514e05 n/a (libglib-2.0.so.0 + 0x82e05)
                                                     #7  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #8  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26666:
                                                     #0  0x00007f406b32e4b6 n/a (libc.so.6 + 0x834b6)
                                                     #1  0x00007f406b330cd0 pthread_cond_wait (libc.so.6 + 0x85cd0)
                                                     #2  0x00007f404df0799e n/a (iris_dri.so + 0x10799e)
                                                     #3  0x00007f404debaf8c n/a (iris_dri.so + 0xbaf8c)
                                                     #4  0x00007f404df078cc n/a (iris_dri.so + 0x1078cc)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26662:
                                                     #0  0x00007f406b32e4b6 n/a (libc.so.6 + 0x834b6)
                                                     #1  0x00007f406b330cd0 pthread_cond_wait (libc.so.6 + 0x85cd0)
                                                     #2  0x00007f404df0799e n/a (iris_dri.so + 0x10799e)
                                                     #3  0x00007f404debaf8c n/a (iris_dri.so + 0xbaf8c)
                                                     #4  0x00007f404df078cc n/a (iris_dri.so + 0x1078cc)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26663:
                                                     #0  0x00007f406b32e4b6 n/a (libc.so.6 + 0x834b6)
                                                     #1  0x00007f406b330cd0 pthread_cond_wait (libc.so.6 + 0x85cd0)
                                                     #2  0x00007f404df0799e n/a (iris_dri.so + 0x10799e)
                                                     #3  0x00007f404debaf8c n/a (iris_dri.so + 0xbaf8c)
                                                     #4  0x00007f404df078cc n/a (iris_dri.so + 0x1078cc)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26655:
                                                     #0  0x00007f406b3a60bf __poll (libc.so.6 + 0xfb0bf)
                                                     #1  0x00007f406b53e1df n/a (libglib-2.0.so.0 + 0xac1df)
                                                     #2  0x00007f406b4e6132 g_main_context_iteration (libglib-2.0.so.0 + 0x54132)
                                                     #3  0x00007f406b4e6182 n/a (libglib-2.0.so.0 + 0x54182)
                                                     #4  0x00007f406b514e05 n/a (libglib-2.0.so.0 + 0x82e05)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26664:
                                                     #0  0x00007f406b32e4b6 n/a (libc.so.6 + 0x834b6)
                                                     #1  0x00007f406b330cd0 pthread_cond_wait (libc.so.6 + 0x85cd0)
                                                     #2  0x00007f404df0799e n/a (iris_dri.so + 0x10799e)
                                                     #3  0x00007f404debaf8c n/a (iris_dri.so + 0xbaf8c)
                                                     #4  0x00007f404df078cc n/a (iris_dri.so + 0x1078cc)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26667:
                                                     #0  0x00007f406b3ab7fd syscall (libc.so.6 + 0x1007fd)
                                                     #1  0x00007f406b537ad7 g_cond_wait_until (libglib-2.0.so.0 + 0xa5ad7)
                                                     #2  0x00007f406b4b5893 n/a (libglib-2.0.so.0 + 0x23893)
                                                     #3  0x00007f406b51a6cb n/a (libglib-2.0.so.0 + 0x886cb)
                                                     #4  0x00007f406b514e05 n/a (libglib-2.0.so.0 + 0x82e05)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     
                                                     Stack trace of thread 26665:
                                                     #0  0x00007f406b32e4b6 n/a (libc.so.6 + 0x834b6)
                                                     #1  0x00007f406b330cd0 pthread_cond_wait (libc.so.6 + 0x85cd0)
                                                     #2  0x00007f404df0799e n/a (iris_dri.so + 0x10799e)
                                                     #3  0x00007f404debaf8c n/a (iris_dri.so + 0xbaf8c)
                                                     #4  0x00007f404df078cc n/a (iris_dri.so + 0x1078cc)
                                                     #5  0x00007f406b3318fd n/a (libc.so.6 + 0x868fd)
                                                     #6  0x00007f406b3b3a60 n/a (libc.so.6 + 0x108a60)
                                                     ELF object binary architecture: AMD x86-64

```

```
# iwgtk -i
Bus name 'org.kde.StatusNotifierItem-26992-1' has been lost
```
Status icon is not visible.

```
$ yay -Qi iwgtk
Name            : iwgtk
Version         : 0.8-2
...
```

`Archlinux`
`XFCE4 4.16.1-3`
```
$ uname -a
Linux ebook 6.0.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 10 Nov 2022 21:14:24 +0000 x86_64 GNU/Linux
```

**Comment by andruska** [comment]
Solved by recommendation on https://wiki.archlinux.org/title/iwd
`Note: As of version 1.23, only root and members of the netdev or the wheel user group are allowed to interact with iwd. In order to use iwctl or iwgtk, you need to add your user to one of those groups.`

**Comment by aagor** [comment]
I just stumbled upon the same issue. But for me this issue shouldn't be closed.
It's fully ok that ` iwgtk`  only works with sufficient permissions, but lack of these permissions should not lead to a Segfault. Instead there should be a suitable error message and an exit with a suitable exit code.

**Comment by andruska** [comment]
The segfault message has been caused by iwgtk a few times now as well, after adding the user to the wheel group.  At the same time, it does not disturb the use very much and the program performs its tasks well.

**Comment by J-Lentz** [comment]
The segfault caused by lack of membership in `netdev` or `wheel` has been fixed in commit 1695a0ba58c65a6c4d43bb4f9a7e18a3910caf0c.

However, the segfaults that you encountered after adding your account to `wheel` are presumably a separate bug.

**Comment by aagor** [comment]
@J-Lentz Thank you for your quick fix.
I will try to get a stacktrace, if the segfault andruska described appears more often.
